flaml/tune/searcher/search_thread.py

```diff
@@ -137,7 +137,10 @@ def on_trial_complete(
         if result:
             self.cost_last = result.get(self.cost_attr, 1)
             self.cost_total += self.cost_last
-            if self._search_alg.metric in result:
+            if self._search_alg.metric in result and (
+                not hasattr(self._search_alg, "lexico_objectives")
+                or self._search_alg.lexico_objectives is None
```

Comment:
```suggestion
                getattr(self._search_alg, "lexico_objectives", None) is None
```
This is a temporary hack, right? Adding this condition might have unwanted effects. For example, if the caller of CFO tries to access the search thread's `cost_best` etc., it will get incorrect results. If it has to be done this way, this behavior needs to be documented. And all the references to CFO need to be checked to make sure when lexico objectives are used, these stale fields are not misused.

Reply:
I add a TODO. This kind of caller only fails when Lexicographic_objectives != None.

Reply:
When lexicographic_objectives is not None, what will go wrong? That needs to be checked throughout automl.py and potentially other files which depend on the affected fields here.

Reply:
I checked.
Fail conditions:
1. When Lexicographic_objectives is None
2. When the algorithm is CFO
3. When we call SearchThread.cost_best, SearchThread.cost_best.obj_best
This only makes __update_speed()_ invalid in LexiFlow and it will not impact other algorithms.